R/create_base_vectors.R

```diff
@@ -1,299 +1,477 @@
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select("MAP_LABEL") |>
+    bcdata::collect() |>
+    dplyr::select("MAP_LABEL")
+
+  if (nrow(bec) > 0) {
+    bec <- sf::st_intersection(bec, aoi)
+    sf::st_write(bec, fs::path(out_dir, "bec.gpkg"), append = FALSE)
+    cli::cat_line()
+    cli::cli_alert_success(
+      "Bec layer downloaded and to written to {.path {out_dir}}"
+    )
+  } else {
+    cli::cli_alert_warning("No BEC data found, check your aoi is within the BEC mapping region")
+  }
+}
+### 2) Download VRI -----------------------
+get_VRI <- function(aoi, out_dir) {
+  cli::cli_alert_info("Downloading VRI layers")
+
+  vri <- bcdata::bcdc_query_geodata("2ebb35d8-c82f-4a17-9c96-612ac3532d55") |>
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select(c("BCLCS_LEVEL_2", "BCLCS_LEVEL_4", "PROJ_AGE_CLASS_CD_1", "SPECIES_CD_1")) |> # Treed sites
+    bcdata::collect() |>
+    dplyr::select(c("BCLCS_LEVEL_2", "BCLCS_LEVEL_4", "PROJ_AGE_CLASS_CD_1", "SPECIES_CD_1"))
+
+  if (nrow(vri) > 0) {
+    vri <- sf::st_intersection(vri, aoi)
+
+    sf::st_write(vri, fs::path(out_dir, "vri.gpkg"), append = FALSE)
+    cli::cat_line()
+    cli::cli_alert_success(
+      "VRI layer downloaded and to written to {.path {out_dir}}"
+    )
+
+    # Subset VRI based on age class and deciduous species
+
+    # class 1 and 2 (0 - 40 yrs)
+
+    vri_class2 <- vri |>
+      dplyr::mutate(age_class = as.numeric(as.character(vri$PROJ_AGE_CLASS_CD_1))) |>
+      dplyr::filter(.data$age_class < 3)
+
+    if (nrow(vri_class2) > 0) {
+      sf::st_write(vri_class2, fs::path(out_dir, "vri_class1_2.gpkg"), append = FALSE)
+      cli::cat_line()
+      cli::cli_alert_success(
+        "VRI class 1 and 2 (0-40 yrs) polygons subset and to written to {.path {out_dir}}"
+      )
+    } else {
+      cli::cat_line()
+      cli::cli_alert_warning("no VRI with class 1 and 2 found")
+    }
+
+    # class 1-3 (0 - 60 years)
+
+    vri_class3 <- vri |>
+      dplyr::mutate(age_class = as.numeric(as.character(vri$PROJ_AGE_CLASS_CD_1))) |>
+      dplyr::filter(.data$age_class == 3)
+
+    if (nrow(vri_class3) > 0) {
+      sf::st_write(vri_class3, fs::path(out_dir, "vri_class3.gpkg"), append = FALSE)
+      cli::cat_line()
+      cli::cli_alert_success(
+        "VRI class 1 to 3 (0-60yrs) polygons subset and to written to {.path {out_dir}}"
+      )
+    } else {
+      cli::cat_line()
+      cli::cli_alert_warning("no VRI with class 1 to 3 found")
+    }
+
+    # deciduous predominant polygons
+
+    vri_decid <- vri |>
+      dplyr::filter(.data$SPECIES_CD_1 %in% c("AT", "EP")) # note might need to adjust for some areas of interest
+
+    if (nrow(vri_decid) > 0) {
+      sf::st_write(vri_decid, fs::path(out_dir, "vri_decid.gpkg"), append = FALSE)
+      cli::cat_line()
+      cli::cli_alert_success("VRI class 1 to 3 (0-60yrs) polygons subset and to written to {.path {out_dir}}")
```

Comment:
```suggestion
      cli::cli_alert_success("deciduous polygons subset and to written to {.path {out_dir}}")
```

Just a missed copy-paste edit I think?